examples/data_assimilation.jl

```diff
@@ -0,0 +1,173 @@
+# # Calibrating a biogeochemical model with `EnsembleKalmanProcesses`
+#
+# In this example we calibrate some of the parameters for the [NPZD](@ref NPZD) model
+# using a data assimilaiton package called [EnsembleKalmanProcesses](https://github.com/CliMA/EnsembleKalmanProcesses.jl).
+# First we setup the model and generate synthetic data with "true" parameters. We then
+# define priors and setup an EKP to solve.
+#
+# While this is a very simple situation it illustraites the ease of integration with 
+# data assimilation tools. Examples given in the EnsembleKalmanProcesses docs illustraite
+# how the package can be used to solve more complex forward models.
+
+# ## Install dependencies
+# First we ensure we have the required dependencies installed
+# ```julia
+# using Pkg
+# pkg "add OceanBioME, Oceananigans, CairoMakie, EnsembleKalmanProcesses, Distributions"
+# ```
+
+using OceanBioME, EnsembleKalmanProcesses, JLD2, CairoMakie, Oceananigans.Units
+using LinearAlgebra, Random
+
+using Distributions
+
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+
+const year = years = 365day
+
+rng_seed = 41
+rng = Random.MersenneTwister(rng_seed)
+
+# Setup the forward model
+
+@inline PAR⁰(t) = 60 * (1 - cos((t + 15days) * 2π / year)) * (1 / (1 + 0.2 * exp(-((mod(t, year) - 200days) / 50days)^2))) + 2
+
+@inline PAR(t) = PAR⁰(t) * exp(- 0.2 * 10)
+
+function model(initial_photosynthetic_slope, 
```

Comment:
This code a little harder to read because this function is called `model`, but there is also an object below called `model = BoxModel(...)`. For examples I think it's good to be abnormally verbose, both for understandability and because a lot of code will end up being based on the example, so it has an outsized influence on coding style in the community.

Reply:
Maybe following the style from other examples, e.g. `simulation = Simulation(...`, you could do `boxmodel = BoxModel(...`.

Reply:
I'd advocate changing the name of the function to precisely reflect what is returned

Reply:
Here, the return values are `P, times`.

Reply:
Thats a good point, thank you!

I will change to something like `run_box_simulation`.